repo/blob/azure/azure_versioned_test.go

```diff
@@ -0,0 +1,227 @@
+package azure_test
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+
+	"github.com/kopia/kopia/internal/clock"
+	"github.com/kopia/kopia/internal/gather"
+	"github.com/kopia/kopia/internal/testlogging"
+	"github.com/kopia/kopia/internal/testutil"
+	"github.com/kopia/kopia/repo/blob"
+	"github.com/kopia/kopia/repo/blob/azure"
+	"github.com/kopia/kopia/repo/format"
+)
+
+func TestGetBlobVersions(t *testing.T) {
+	t.Parallel()
+	testutil.ProviderTest(t)
+
+	// must be with Immutable Storage with Versioning enabled
+	container := getEnvOrSkip(t, testContainerEnv)
```

Comment:
Use the `KOPIA_AZURE_TEST_IMMUTABLE_*` variant of the environment variables for these tests. We'll run the provider tests before merging to make sure the tests won't break.

Reply:
updated

Reply:
Mike: apologies, my previous comment was ambiguous.
Use the `KOPIA_AZURE_TEST_IMMUTABLE_*` variant of the environment variables **only** in the new tests that are added here.

Reply:
updated